code/modules/mob/living/living_defense.dm

```diff
@@ -257,24 +257,44 @@
 	spawn(1) updatehealth()
 	return 1
 
-/mob/living/proc/IgniteMob()
+/mob/living/proc/IgniteMob(var/fire_stacks_to_add = 0)
+
+	if(fire_stacks_to_add)
+		adjust_fire_stacks(fire_stacks_to_add)
+
 	if(fire_stacks > 0 && !on_fire)
 		on_fire = 1
 		set_light(light_range + MOB_FIRE_LIGHT_RANGE, light_power + MOB_FIRE_LIGHT_POWER)
 		update_fire()
+		return TRUE
+
+	return FALSE
 
-/mob/living/proc/ExtinguishMob()
-	if(on_fire)
+/mob/living/proc/ExtinguishMob(var/fire_stacks_to_remove = 0)
+
+	adjust_fire_stacks(-fire_stacks_to_remove)
+
+	if(fire_stacks <= 0 && on_fire)
 		on_fire = 0
-		fire_stacks = 0
 		set_light(max(0, light_range - MOB_FIRE_LIGHT_RANGE), max(0, light_power - MOB_FIRE_LIGHT_POWER))
 		update_fire()
+		return TRUE
+
+	return FALSE
+
+/mob/living/proc/ExtinguishMobCompletely()
+	return ExtinguishMob(fire_stacks)
 
 /mob/living/proc/update_fire()
 	return
 
-/mob/living/proc/adjust_fire_stacks(add_fire_stacks) //Adjusting the amount of fire_stacks we have on person
-    fire_stacks = Clamp(fire_stacks + add_fire_stacks, FIRE_MIN_STACKS, FIRE_MAX_STACKS)
+/mob/living/proc/adjust_fire_stacks(var/add_fire_stacks)
+	if(!fire_stacks)

```

Comment:
This can be removed. `null + n` where `n` is a number is safe and will result in a number.